src/installer.ts

```diff
@@ -100,9 +101,17 @@ async function queryLatestMatch(versionSpec: string): Promise<string> {
 
   let versions: string[] = [];
   let dataUrl = 'https://nodejs.org/dist/index.json';
-  let rest: restm.RestClient = new restm.RestClient('setup-node');
-  let nodeVersions: INodeVersion[] =
-    (await rest.get<INodeVersion[]>(dataUrl)).result || [];
+  let httpClient = new hc.HttpClient('setup-node', [], {
```

Comment:
i added retries here when downloading the versions JSON

further below when the exe is downloaded, `toolCache.downloadTool` already has retries built-in